### PR TITLE
Server container documentation

### PR DESCRIPTION
**Pull request by mbussolotto**
DO NOT MERGE YET: Awaiting container server part to be merged, and the confirmation of the package name.

# Description

Installation/Upgrade Guide -> Installation -> Uyuni Server
Install Containerized Uyuni Server
1. Container Host Requirements
2. Install Container Services on the host system
3. Customize Uyuni Server configuration
4. Allow network access for provided services on container host firewall

Installation/Upgrade Guide -> Setup -> Uyuni Server
Containerized Uyuni Server Setup
1. Set Uyuni Server parameters
2. Install uyuni-tools

Reference Guide -> Command Line Tools -> Container CLI Tools
Uyuni Tools
1. Supportconfig generation

# Target branches

Which documentation version does this PR apply to?

- [x] Master (Default)
- [ ] Manager-4.3
- [ ] Manager-4.2

# Links

Fixes https://github.com/SUSE/spacewalk/issues/20055

### CODE REVIEW

File: modules/installation-and-upgrade/pages/server-container-installation.adoc [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+|
+| Minimum 2 dedicated 64-bit CPU cores
+
+| RAM
+| Test Server
+| Minimum 2{nbsp}GB
+
+|
+| Production Server
+| Minimum 8{nbsp}GB
```
Comment (cbosdo) [inline modules/installation-and-upgrade/pages/server-container-installation.adoc:38]: 8GB is not enough due to missing swap.

File: modules/reference/pages/cli-container.adoc [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+== Description
+The [package]``uyuni-tools`` package has components as subpackages with CLI tools:
+
+[[uyuni-tools-components]]
+.Uyuni Tools CLI Tool Components
+[cols="1,1,1,1", options="header"]
+|===
+| Package name            | Tool name               | Installed on | Use for
+| ``uyuni-tools``         | ``uyunictl``            | Host         | day-to-day operations against a containerized {productname} server
```
Comment (cbosdo) [inline modules/reference/pages/cli-container.adoc:15]: Keep in mind that the package doesn't exist yet. We'll need to make sure this is the final name

File: modules/installation-and-upgrade/pages/server-container-installation.adoc [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+====
+
+
+
+[[installation-server-containers-services]]
+== Install Container Services on the host system
+
+{productname} Server containers are using [literal]``podman`` and [literal]``systemd`` to run and manage all server containers.
+
+First step is to install container control files provided by package [literal]``uyuni-server-systemd-services``.
```
Comment (keichwa) [inline modules/installation-and-upgrade/pages/server-container-installation.adoc:59]:
```suggestion
First step is to install a base system with the container control files provided by the [package]``uyuni-server-systemd-services`` package.
```

File: modules/installation-and-upgrade/pages/server-container-installation.adoc [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
++
+----
+zypper ref
+----
+. Install the pattern for the {productname} Server as [systemitem]``root``:
++
+----
+zypper install uyuni-server-systemd-services
+----
+
```
Comment (keichwa) [inline modules/installation-and-upgrade/pages/server-container-installation.adoc:87]:
```suggestion



```

File: modules/installation-and-upgrade/pages/server-container-installation.adoc [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+- 4505/TCP - Salt
+- 4506/TCP - Salt
+- 5432/TCP - PostgreSQL
+- 8022/TCP - SSH
+- 9100/TCP - Prometheus node exporter
+- 9187/TCP - Prometheus PostgreSQL exporter
+- 9800/TCP - Promethues Metrics Exporter
+- 25151/TCP - Cobbler
+
+Continue with setting up the installed {productname} server as a containers at xref:server-container-setup.adoc[].
```
Comment (keichwa) [inline modules/installation-and-upgrade/pages/server-container-installation.adoc:155]:
```suggestion
Continue with setting up the installed {productname} server as a container at xref:server-container-setup.adoc[].
```

File: modules/installation-and-upgrade/pages/server-container-setup.adoc [hunk 6]
```diff
@@ -0,0 +1,89 @@
+[[server-setup-containers]]
```
Comment (keichwa) [inline modules/installation-and-upgrade/pages/server-container-setup.adoc:1]:
```suggestion
[[server-container-setup]]
```

File: modules/installation-and-upgrade/pages/server-container-setup.adoc [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+= Containerized {productname} Server Setup
+
+Once container host for {productname} Server containers is prepared, setup of containers require few additional steps to finish configuration.
```
Comment (keichwa) [inline modules/installation-and-upgrade/pages/server-container-setup.adoc:4]:
```suggestion
When the container host for {productname} Server container is prepared, setup of the container requires a few additional steps to finish the configuration.
```

File: modules/installation-and-upgrade/pages/server-container-setup.adoc [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+
+. In the host system
++
+----
+systemctl start uyuni-server.service 
+----
+
+[IMPORTANT]
+====
+{productname} Server setup take times. You can get logs from the ```journalctl -xeu uyuni-server.service``` services using ```journalctl```.
```
Comment (keichwa) [inline modules/installation-and-upgrade/pages/server-container-setup.adoc:46]:
```suggestion
{productname} Server setup takes time. You can see the logs with [command]``journalctl -xeu uyuni-server.service``.
```

File: modules/installation-and-upgrade/pages/server-container-setup.adoc [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+You can also use `podman logs` using the same name.
+====
+
+[[migrate-server-uyuni]]
+== Migrate {productname} server to container
+
+Migrate a {productname} server installation requires some additional steps:
```
Comment (keichwa) [inline modules/installation-and-upgrade/pages/server-container-setup.adoc:53]:
```suggestion
Migrate a {productname} server installation requires additional steps:
```

File: modules/installation-and-upgrade/pages/server-container-setup.adoc [hunk 10]
```diff
@@ -0,0 +1,89 @@
...
+----
+git clone --depth 1 https://github.com/PROJECT/uyuni-tools /root/uyuni-tools
+----
+. Compile and install it
++
+----
+cd ~/uyuni-tools && go build -o ./bin ./... && ln -s /root/uyuni-tools/bin/uyunictl /usr/bin/uyunictl
+----
+
+More information about [package]``uyuni-tools`` and its usage at xref:reference:cli-container.adoc[CLI Container Tools].
```
Comment (keichwa) [inline modules/installation-and-upgrade/pages/server-container-setup.adoc:89]:
```suggestion
For more information about [package]``uyuni-tools`` and its usage, see xref:reference:cli-container.adoc[].
```

File: modules/reference/pages/cli-container.adoc [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+----
++
+
+. Copy supportconfig in your host system
++
+----
+uyunictl cp server:SUPPORTCONFIG_CONTAINER_PATH SUPPORTCONFIG_HOST_PATH
+----
++
+where [[path]]```SUPPORTCONFIG_HOST_PATH``` is the destination path in your host system.
```
Comment (keichwa) [inline modules/reference/pages/cli-container.adoc:39]:
```suggestion
where [path]``SUPPORTCONFIG_HOST_PATH`` is the destination path in your host system.
```

**Review by keichwa — COMMENTED** [review]
Thanks for adding documentation!  A few mostly just cosmetic suggestions.

File: modules/installation-and-upgrade/pages/server-container-installation.adoc [hunk 12]
```diff
@@ -0,0 +1,156 @@
...
+
+
+
+[[installation-server-containers-customize-config]]
+== Customize {productname} Server configuration
+
+{productname} Server containers require some volumes to be mounted for long term storage.
+Those volumes are automatically created by podman and can be listed using the [literal]``podman volume ls`` command.
+By default, [literal]``podman`` stores the files of the volumes in [path]``/var/lib/containers/storage/volumes``.
+The needed volume are named:
```
Comment (keichwa) [inline modules/installation-and-upgrade/pages/server-container-installation.adoc:96]:
```suggestion
The needed volumes are named:
```

File: modules/reference/pages/cli-container.adoc [hunk 13]
```diff
@@ -0,0 +1,39 @@
+[[ref-cli-uyuni-tools]]
+= Uyuni Tools
+
+
+[[uyuni-tools-description]]
+
```
Comment (keichwa) [inline modules/reference/pages/cli-container.adoc:6]:
```suggestion


[[uyuni-tools-description]]
```

File: modules/reference/pages/cli-container.adoc [hunk 14]
```diff
@@ -0,0 +1,39 @@
...
+|===
+| Package name            | Tool name               | Installed on | Use for
+| ``uyuni-tools``         | ``uyunictl``            | Host         | day-to-day operations against a containerized {productname} server
+| ``uyuni-tools``         | ``uyuniadm``            | Host         | helps deployment and setup of a containerized {productname} server
+|===
+
+
+
+[[uyuni-tools-usage]]
+== Usate
```
Comment (keichwa) [inline modules/reference/pages/cli-container.adoc:22]:
```suggestion
== Usage
```

**Review by keichwa — APPROVED** [review]
Just cosmetic suggestions or typos.

File: modules/reference/pages/cli-container.adoc [hunk 15]
```diff
@@ -0,0 +1,108 @@
...
+      --server string              Hostname of the source server
+      --ssh_auth_socket string     SSH_AUTH_SOCKET value
+      --tag string                 Tag Image (default "latest")
+      --user string                User used to login in source server
+----
+
+[[uyuni-tools-supportconfig]]
+=== Supportconfig generation
+
+. In the host system run the command for generating supportconfig
```
Comment (keichwa) [inline modules/reference/pages/cli-container.adoc:95]:
```suggestion
. In the host system, run the command for generating supportconfig:
```

**Review by keichwa — COMMENTED** [review]
Most just cosmetic suggestions.

**Comment by mbussolotto** [comment]
@keichwa Thanks for the suggestion! Please consider that the PR is quite old, so I'm sure something is changed in the meanwhile, but since container changes are in master we should tackle them soon :) 

cc @cbosdo

File: modules/installation-and-upgrade/pages/server-container-setup.adoc [hunk 16]
```diff
@@ -0,0 +1,111 @@
...
+
+Migrate a {productname} server installation means transfer all the functionality from a source server to a destination server that will host a productname} server container. 
+
+Prerequisites are:
+
+* ``UYUNI_FQDN`` in ``/etc/sysconfig/uyuni-server-systemd-services`` should be the same of the source address.
+* The destination server user should be able to `rsync` the source server without password. 
+* The destination server should provide a ssh-agent and the `SSH_AUTH_SOCK` should be set for the desired user.
+
+.Procedure: Enable passwordless `rsync`
```
Comment (0rnela) [inline modules/installation-and-upgrade/pages/server-container-setup.adoc:65]:
```suggestion
.Procedure: Enabling passwordless `rsync` 
```

File: modules/installation-and-upgrade/pages/server-container-setup.adoc [hunk 17]
```diff
@@ -0,0 +1,111 @@
...
+If not, run `ssh-keygen` to generate it.
+If the source server user is not root, it should be able to run [command]``rsync``.
+It can be done by adding to [path]``/etc/sudoers``:
++
+```
+add to sudoers file
+$SOURCE_USER ALL=(ALL) NOPASSWD:/usr/bin/rsync
+```
+
+.Procedure: Provide an ssh agent and add key
```
Comment (0rnela) [inline modules/installation-and-upgrade/pages/server-container-setup.adoc:94]:
```suggestion
.Procedure: Providing ssh agent and adding key
```

File: modules/installation-and-upgrade/pages/server-container-installation.adoc [hunk 18]
```diff
@@ -0,0 +1,156 @@
...
+[[installation-server-containers-services]]
+== Install container services on the host system
+
+{productname} Server containers are using [literal]``podman`` and [literal]``systemd`` to run and manage all server containers.
+
+First step is to install a base system with the container control files provided by the [package]``uyuni-server-systemd-services`` package.
+
+
+[[proc-installation-server-containers-services]]
+.Procedure: Installation of container services for {productname} Server
```
Comment (0rnela) [inline modules/installation-and-upgrade/pages/server-container-installation.adoc:63]:
```suggestion
.Procedure: Installing the container services for {productname} Server
```

**Review by 0rnela — APPROVED** [review]

**Comment by jcayouette** [comment]
I am curious as to what these docs are covering? I am already working on Installation container docs?

**Comment by jcayouette** [comment]
Do not merge this please. I am working on docs and will include anything missed here in the up-to-date content.

**Comment by cbosdo** [comment]
> Do not merge this please. I am working on the container installation docs and will include everything from here in the up-to-date content.

It's a very old draft of what you are currently doing. Given how things have changed in 9 months we surely can drop this PR

**Comment by jcayouette** [comment]
Yes, I only used it for review. Closing

**Pull request closed without merging**